pkg/controller/machineset/machineset_controller.go

```diff
@@ -230,12 +245,188 @@ func (c *Controller) processNextWorkItem() bool {
+		return err
+	}
+
+	ms, err := c.machineSetsLister.MachineSets(namespace).Get(name)
+	if errors.IsNotFound(err) {
+		c.logger.WithField("key", key).Errorf("JDIAZ couldn't find machineset %v", key)
+		return err
+	}
+	if err != nil {
+		c.logger.Errorf("JDIAZ problem fetching machineset %v", key)
+		return err
+	}
+	if ms.Spec.NodeType == "Master" {
+		// masters aren't cattle, they need ansible depoyment
+		c.logger.WithField("machineset", ms).Debugf("JDIAZ ignoring master machineset: %v", ms)
+		return nil
+	}
+	//c.logger.WithField("machineset", ms).Debugf("JDIAZ full machineset: %+v", ms)
+
+	// check DeletionTimestamp???
+
+	// get all the machines in our current namespace
+	allMachines, err := c.machineLister.Machines(namespace).List(labels.Everything())
+	if err != nil {
+		c.logger.WithField("machineset", ms).Errorf("JDIAZ error while trying to get list of all machines")
+		return err
+	}
+	c.logger.WithField("machineset", name).Debugf("JDIAZ machinelist: %v", allMachines)
+
+	// get all machines which are children of this machineset
+	var filteredMachines []*clusteroperator.Machine
+	for _, machine := range allMachines {
+		if machine.DeletionTimestamp != nil {
+			continue
+		}
+		controllerRef := metav1.GetControllerOf(machine)
+		if controllerRef == nil {
+			c.logger.WithField("machineset", name).Debugf("JDIAZ no controller ref for machine %v", machine)
+			continue
+		}
+		if ms.UID != controllerRef.UID {
+			//c.logger.WithField("machineset", name).Debug("JDIAZ the kid is not my son %v", machine)
+			continue
+		}
+		filteredMachines = append(filteredMachines, machine)
+	}
+	//c.logger.WithField("machineset", name).Debugf("JDIAZ children machines: %v", filteredMachines)
+
+	var manageMachinesErr error
+
+	//if machineSetNeedsSync && ms.DeletionTimestamp == nil {
+	if true && ms.DeletionTimestamp == nil {
+		manageMachinesErr = c.manageMachines(filteredMachines, ms)
+	}
+
+	if manageMachinesErr != nil {
+		return manageMachinesErr
+	}
+
+	return nil
+}
+
+func (c *Controller) manageMachines(machines []*clusteroperator.Machine, ms *clusteroperator.MachineSet) error {
+	/*machineSetKey, err := controller.KeyFunc(ms)
+	if err != nil {
+		// FIXME
+		return nil
+	}*/
+
+	machinePrefix := ms.ObjectMeta.Name + "-"
+	//c.logger.WithField("machineset", ms.ObjectMeta.Name).Debugf("JDIAZ machine prefix: %v", machinePrefix)
+
+	machineSetMachines := []*clusteroperator.Machine{}
+	//machinesToCreate := []*clusteroperator.Machine{}
+	machinesToDelete := []*clusteroperator.Machine{}
+	//machinesToUpdate := []*clusteroperator.Machine{}
+
+	deleteCount := int(0)
+	createCount := int(0)
+
+
+	// caculate how many to create/delete
+	if len(machines) > ms.Spec.Size {
+		deleteCount = len(machines) - ms.Spec.Size
+	}
+	if len(machines) < ms.Spec.Size {
+		createCount = ms.Spec.Size - len(machines)
+	}
+
+	for _, machine := range machines {
+		if ! strings.HasPrefix(machine.ObjectMeta.Name, machinePrefix) {
```

Comment:
I don't think we need to worry about validating the machine name. We are using the owner reference as the determination of to which machine set the machine belongs. The name is just a convenience for the human user.